### DISCUSSION TITLE
Front Camera

### DISCUSSION THREAD

**Discussion by harishcmathur** [tesla-android/issue-tracker#253] [Ideas]
Hi,  Any plan to add a front parking camera to this?  Perhaps one that pops up when in drive and moving below 5 miles/hr?  Particularly given Tesla has removed the front sensors.

**Comment by mikegapinski** [comment]
You could theoretically use a USB camera with Android, but I doubt it would be a fun experience, especially if you consider having to get the cable through the firewall

**Reply by harishcmathur** [reply]
Wow,  that was a fast reply!   I would not mind going through the work to install a camera.  I watched a few you-tube videos and does look like it will take some effort, but doable.  I think you will find others that agree... given that there are no longer parking sensors on the front of the car (I just picked up a model Y). Any recommendations on packages or how I could proceed?  It's been a long time since I wrote any code (think PASCAL ;-)) So likely would be better off installing existing packages.  I'm going to start gathering parts for Tesla Android and looking forward to using it. Thank you so much for your work!

**Reply by mikegapinski** [reply]
There are things on the horizon that would make things like what you mentioned within reach...

I have OBD working via the S3XY Buttons Commander module, currently via BT for their dashboard. I am hoping to iterate on this once I have the time and actually integrate with the CAN network. I don't want to wire Tesla Android directly to CAN in the center console or near the MCU. This is a security issue I don't want to deal with, the Commander is secure, BT pairing requires physical access to the car(tapping the buttons on the wheel when new device is added) and Wi-Fi is password protected.

When it comes to the cameras, there is one user who wants to use Tesla Android for video conferences in Drive. You pair the device with the car via BT for microphone access and connect a small camera that can be used for video optionally. It's not yet fully working but if it does it might take us somewhere when it comes to the bumper cameras.

BTW... I feel for you, I drove the Model Y without USS and the Vision system is a joke, and it wasn't even raining. Tesla had one of the best implementations of USS on the market, it's super reliable and very accurate :/

**Comment by tw02d008** [comment]
I am as well very intressed in a camera solution. Model 3 without uss -.- 
I think something like a wifi camera in front would be nice and than somehow to connect via vlc to the videostream.  Or maybe some camera with a dedicated android app?  but i guess this would be a problem because such cameras create their own wifi network. is it possible to connect a usb wlan adapter to the raspi and join/create two networks?
